cmake/TestSetup.cmake

```diff
@@ -108,52 +108,58 @@ endfunction()
 # test output will be sent to /tst/regression/outputs/dir_mpi
 # test property labels: regression, mpi-yes
 function(setup_test_mpi nproc dir arg extra_labels)
-  if( MPI_FOUND )
-    separate_arguments(arg) 
-    list(APPEND labels "regression;mpi-yes")
-    list(APPEND labels "${extra_labels}")
+  if(ENABLE_MPI)
+    if( MPI_FOUND)
+      separate_arguments(arg) 
+      list(APPEND labels "regression;mpi-yes")
+      list(APPEND labels "${extra_labels}")
 
-    if(Kokkos_ENABLE_CUDA)
-      set(PARTHENON_KOKKOS_TEST_ARGS "--kokkos-num-devices=${NUM_GPU_DEVICES_PER_NODE}")
-      list(APPEND labels "cuda")
-    endif()
-    if (Kokkos_ENABLE_OPENMP)
-      set(PARTHENON_KOKKOS_TEST_ARGS "${PARTHENON_KOKKOS_TEST_ARGS} --kokkos-threads=${NUM_OMP_THREADS_PER_RANK}")
+      if(Kokkos_ENABLE_CUDA)
+        set(PARTHENON_KOKKOS_TEST_ARGS "--kokkos-num-devices=${NUM_GPU_DEVICES_PER_NODE}")
+        list(APPEND labels "cuda")
+      endif()
+      if (Kokkos_ENABLE_OPENMP)
+        set(PARTHENON_KOKKOS_TEST_ARGS "${PARTHENON_KOKKOS_TEST_ARGS} --kokkos-threads=${NUM_OMP_THREADS_PER_RANK}")
+      endif()
+      process_mpi_args(${nproc})
+      add_test( NAME regression_mpi_test:${dir} COMMAND ${Python3_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/run_test.py
+        ${MPIARGS} ${arg}
+        --test_dir ${CMAKE_CURRENT_SOURCE_DIR}/test_suites/${dir}
+        --output_dir "${PROJECT_BINARY_DIR}/tst/regression/outputs/${dir}_mpi"
+        --kokkos_args=${PARTHENON_KOKKOS_TEST_ARGS})
+      set_tests_properties(regression_mpi_test:${dir} PROPERTIES LABELS "${labels}" RUN_SERIAL ON )
+      record_driver("${arg}")
+    else()
+      message(STATUS "MPI not found, not building regression tests with mpi")
```

Comment:
One more question:
When does the new additional conditional actually kick it?
According to our main CMakeLists.txt
```
set(ENABLE_MPI OFF)
set(NUM_MPI_PROC_TESTING "4" CACHE STRING "Number of mpi processors to use when running tests with MPI")
if (NOT PARTHENON_DISABLE_MPI)
  find_package(MPI COMPONENTS CXX)
  if (NOT MPI_FOUND)
    message(FATAL_ERROR "MPI is required but couldn't be found. "
    "If you want to build Parthenon without MPI, please rerun CMake with -DPARTHENON_DISABLE_MPI=ON")
  endif()
  set(ENABLE_MPI ON)
endif()
```
ENABLE_MPI=ON implies that MPI_FOUND is true, otherwise it would have failed already earlier.

Reply:
Yes, ENABLE_MPI=ON implied MPI_FOUND is true, but just because MPI is found does not mean the user wants to  compile with it. Maybe I missing something, but when I was just trying to compile with only cuda support on RZAnsel this code block was getting executed and causing problems.

Reply:
I see, I guess I practically never compile without MPI and this didn't come across.
I assume the the problem is that I missed that case in the Summit example.
If "mpi" is not in the string then `PARTHENON_DISABLE_MPI=ON` should be set and everything works as expected.
I'll update the Summit file including the docs shortly.